src/CommandingOfficers/Modifiers/UnitRemodelModifier.java

```diff
@@ -46,17 +42,11 @@ public void applyChanges(Commander commander)
   {
     for( Unit unit : commander.units )
     {
-      if( modelSwaps.containsKey(unit.model.type) )
+      if( modelSwaps.containsKey(unit.model) )
       {
-        // Store off the unit's weapons.
-        unitWeapons.put(unit, unit.weapons);
-
         // Swap the unit's identity and store it for later.
         modelSwapBacks.put(unit, unit.model);
-        unit.model = unit.CO.unitModels.get(modelSwaps.get(unit.model.type));
-
-        // Change out weapons.
-        doWeaponSwap(unit, unit.model);
+        unit.model = modelSwaps.get(unit.model);
```

Comment:
There is no longer any weapon/ammo handling here. Is that just unnecessary now that ammo is part of Unit?

Reply:
Yep. `Weapon`s don't exist, only `WeaponModel`s... which are owned by the `UnitModel`.
Simplifying logic is great.

Also, not requiring a 1:1 mapping between any enums and `UnitModel`s means you could have a CO with multiple forms of a given unit type (say, balancing unit spam by giving them awful attack/defense).